reactioncommerce/reaction
# Set custom layout to override based on priority field

Resolves #1969 

**How to Test:**
- Create a custom plugin with a layout that you want to take precedence over another existing default layout. An easy way is to drop [reaction-example-plugin](https://github.com/reactioncommerce/reaction-example-plugin) into your project.
- Default `priority` is now 999.... so set a lower value for `priority` for your custom layout in the `register.js`
- Run `reaction reset -n && reaction`
- Load the app in the browser and confirm that your defined custom template is used over the default

**Changes introduced in this PR include (do take a look at the diff)**:
- Sorting of the layout array based on the priority field (before picking a match in selectLayout (client/modules/router/main.js) 
- Updated other registry entries using the old value of 1 to 999. This only applies to imports/plugins/core/checkout/register.js
- Added migration for templates already in db, to the new default 999 (4_update_templates_priority.js)
- Updated the `audience` field in Layout schema. It is being used as array of strings in the app... but schema was set as array of objects.
